neuroptica/nonlinearities.py

```diff
@@ -402,3 +405,156 @@ def df_dRe(self, a: np.ndarray, b: np.ndarray) -> np.ndarray:
 
     def df_dIm(self, a: np.ndarray, b: np.ndarray) -> np.ndarray:
         return (a > 0) * (b > 0) * 1j
+
+class TrainableNonLinearity(ComplexNonlinearity):
+    '''
+    Nonlinearity with a trainable parameter
+    '''
+
+    def __init__(self, N, holomorphic, mode):
+        super().__init__(N, holomorphic, mode)
+
+    def compute_gradients(self, inputs: np.ndarray, delta:np.ndarray):
+        '''
+        Compute the gradients for nonlinearity in this layer, without adjusting the parameters
+        :param forward_field: forward-propagating input electric field at the beginning of the nonlinearity
+        :param adjoint_field: backward-propagating output electric field at the end of the nonlinearity
+        :return:
+        '''
+        raise NotImplementedError
+
+
+class FabryPerotInferometer(TrainableNonLinearity):
+    '''
+    CMOS light detector actuated FBI
+    '''
+
+    def __init__(self, N, KAPPA, INIT_VAR):
```

Comment:
please add docstring indicating what `KAPPA` and `INIT_VAR` are. also, please change `KAPPA` and `INIT_VAR` to lowercase to be more consistent with the rest of the codebase (even though it doesn't follow standard PEP8 because math notation, I reserve all-caps multi-letter variables for constants)